modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/display/context/KBArticleItemSelectorViewDisplayContext.java

```diff
@@ -0,0 +1,635 @@
+import com.liferay.portal.kernel.service.GroupLocalServiceUtil;
+import com.liferay.portal.kernel.service.GroupServiceUtil;
+import com.liferay.portal.kernel.servlet.taglib.ui.BreadcrumbEntry;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.JavaConstants;
+import com.liferay.portal.kernel.util.LinkedHashMapBuilder;
+import com.liferay.portal.kernel.util.OrderByComparator;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+import javax.portlet.PortletException;
+import javax.portlet.PortletRequest;
+import javax.portlet.PortletResponse;
+import javax.portlet.PortletURL;
+import javax.portlet.RenderResponse;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Alicia García
+ */
+public class KBArticleItemSelectorViewDisplayContext {
+
+	public KBArticleItemSelectorViewDisplayContext(
+		HttpServletRequest httpServletRequest,
+		InfoItemItemSelectorCriterion infoItemItemSelectorCriterion,
+		String itemSelectedEventName, PortletURL portletURL, boolean search) {
+
+		_httpServletRequest = httpServletRequest;
+		_infoItemItemSelectorCriterion = infoItemItemSelectorCriterion;
+		_itemSelectedEventName = itemSelectedEventName;
+		_portletURL = portletURL;
+		_search = search;
+
+		_portletRequest = (PortletRequest)httpServletRequest.getAttribute(
+			JavaConstants.JAVAX_PORTLET_REQUEST);
+		_portletResponse = (RenderResponse)httpServletRequest.getAttribute(
+			JavaConstants.JAVAX_PORTLET_RESPONSE);
+		_themeDisplay = (ThemeDisplay)_httpServletRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+	}
+
+	public String getDisplayStyle() {
+		if (Validator.isNotNull(_displayStyle)) {
+			return _displayStyle;
+		}
+
+		_displayStyle = SearchDisplayStyleUtil.getDisplayStyle(
+			_httpServletRequest, KBPortletKeys.KNOWLEDGE_BASE_ARTICLE,
+			"item-selector-display-style", "descriptive");
+
+		return _displayStyle;
+	}
+
+	public String getGroupCssIcon(long groupId) throws PortalException {
+		Group group = GroupServiceUtil.getGroup(groupId);
+
+		return group.getIconCssClass();
+	}
+
+	public String getGroupLabel(long groupId, Locale locale)
+		throws PortalException {
+
+		Group group = GroupServiceUtil.getGroup(groupId);
+
+		return group.getDescriptiveName(locale);
+	}
+
+	public Map<String, Object> getKBArticleContext(KBArticle kbArticle) {
+		return HashMapBuilder.<String, Object>put(
+			"returnType", InfoItemItemSelectorReturnType.class.getName()
+		).put(
+			"value",
+			() -> JSONUtil.put(
```

Comment:
Maybe you can get this value calling `getKBArticleDataValue(KBArticle)`

Reply:
I will delete the `getKBArticleContext` method because is not longer needed